Add comments to sb init stories template

kylegach opened the pull request:
## What I did

- Add helpful comments to particular parts of the template files used to generate stories with `npx sb init`
- Only done for frameworks that display in the framework switcher
- URLs in comments are framework-specific
- Other minor tweaks
    - Fix (and alpha-order) argTypes definitions
    - Use "Template" name/concept consistently

~Submitting a draft PR to start the conversation on whether this is a good idea and if this is the right execution. If we come to an agreement on that, I'll propagate this pattern throughout other template files.~ _Done._

## How to test

_Not sure how to test `sb init`, but I don't think these changes would require testing anyway_

- [ ] Is this testable with Jest or Chromatic screenshots?
- [ ] Does this need a new example in the kitchen sink apps?
- [ ] Does this need an update to the documentation?

If your answer is yes to any of these, please make sure to include it in your PR.

<!--

Everybody: Please submit all PRs to the `next` branch unless they are specific to the current release. Storybook maintainers cherry-pick bug and documentation fixes into the `master` branch as part of the release process, so you shouldn't need to worry about this.

Maintainers: Please tag your pull request with at least one of the following:
`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`

-->

shilman reviewed (approved):
Nice one @kylegach

jonniebigodes reviewed (approved):
@kylegach thanks for this. It looks awesome. Appreciate the time you took to add this small but invaluable piece of information to the examples, helping users get more context on the features.

kylegach committed:
Add comments to sb init stories templates

- Fix (and alpha-order) argTypes definitions
- Use "Template" name/concept consistently

kylegach commented:
I force-pushed to add these comments (and some other, minor changes) to the other template files. Please review once more and then feel free to merge. Thanks!
